### PR TITLE
Added a solution for Project Euler Problem 203 "Squarefree Binomial Coefficients"

### PR DESCRIPTION
**Pull request by fernandobperezm**
### **Describe your change:**

Added a solution to Project Euler Problem 203 "Squarefree Binomial Coefficients" [Link](https://projecteuler.net/problem=203).

The solution is based on three main pilars.
1. Calculate the unique coefficients of a Pascal's Triangle of depth `d` .
2. Calculate the prime numbers between `2` and the maximum coefficient `Cmax` using a variant of the Sieve of Eratosthenes [Link](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes) and considering that the square of each prime must be less or equal than that `Cmax`. The calculation returns the square of those primes.
3. Calculate all squarefree numbers by decomposing each number `n` into `n = p * p * r` where `p` is a prime number calculated before and `r` is a positive integer. If no `r` can be found for all squared primes, then the number is squarefree, else, the number is non-squarefree.

After all unique squarefree numbers are calculated, they're summed-up to provide the final answer.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_203/sol1.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    coefficients = set()
+    previous_coefficients = []
+    for step in range(1, depth + 1):
+        if step == 1:
+            coefficients.add(1)
+            previous_coefficients = [1]
+        else:
+            coefficients_begins_one = previous_coefficients + [0]
+            coefficients_ends_one = [0] + previous_coefficients
+            previous_coefficients = []
```
Comment (dhruvmanila) [inline project_euler/problem_203/sol1.py:61]:
```suggestion
    coefficients = {1}
    previous_coefficients = [1]
    for step in range(2, depth + 1):
        coefficients_begins_one = previous_coefficients + [0]
        coefficients_ends_one = [0] + previous_coefficients
        previous_coefficients = []
```
Please test whether this works or not.

Comment (fernandobperezm) [reply project_euler/problem_203/sol1.py:61]: Tested locally and works. Also black, flake8 and doctests pass. Should I commit locally or use the "commit suggestion" button?

File: project_euler/problem_203/sol1.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+    >>> get_squarefree({1}, [])
+    set()
+    >>> get_squarefree({1, 2}, [])
+    set()
+    >>> get_squarefree({1, 2, 3, 4, 5, 6, 7, 35, 10, 15, 20, 21}, [4, 9, 25])
+    {1, 2, 3, 5, 6, 7, 35, 10, 15, 21}
+    """
+
+    def get_squared_primes_to_use(
```
Comment (dhruvmanila) [inline project_euler/problem_203/sol1.py:117]: A function within a function is mostly used as a wrapper. This doesn't seem like that so please separate them out.

**Comment by joshuaonazi** [comment]
can you please help me in participating in the 2020 hacktoberfest, submitting the preferred PR and contributing as required by hacktoberfest thereby being eligible for the T-shirt and swags or planting a tree

**Comment by jratnani20** [comment]
hey @fernandobperezm i am intrested  in doing it ..could uh plz assign it to me ?

File: project_euler/problem_203/sol1.py (hunk 2 above)
Comment (fernandobperezm) [reply project_euler/problem_203/sol1.py]: Done, thanks for the advice!

Taking out the function also revealed that one doctest of `get_squared_primes_to_use` was failing. I fixed it and revisited the function to gracefully handle that failing case.

**Comment by fernandobperezm** [comment]
@jratnani20 @joshuaonazi I'm really sorry I did not see your messages before, this week was really busy for me and I wasn't active here. Also, I just pushed the changes requested by the reviewer :-(

**Review by dhruvmanila — APPROVED** [review]
Look perfect 👌 
Thank you for your contribution! 😄

**Pull request merged**